### PR TITLE
Use bytes for import/export buffer

### LINKED ISSUE #1013: import_attribute_configuration_buffer and export_attribute_configuration_buffer should use `bytes`
#Description of issue

The API for `import_attribute_configuration_buffer` and `export_attribute_configuration_buffer` in several nimi-python APIs uses list of `int` to represent the buffer. This is inefficient and sematically incorrect.

It should use `bytes` in Python 3 and arguably `str` in Python 2.

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [X] I've ~~added~~modified tests applicable for this pull request

### What does this Pull Request accomplish?
* `export_attribute_configuration_buffer()` returns `array.array('b')` instead of list of integers
* `import_attribute_configuration_buffer()` can now accept:
    * list of integers (all less than 255)
    * `array.array('b')`
    * `bytes` object
    * `bytearray` object

### List issues fixed by this Pull Request below, if any.
* Fix #1013 

### What testing has been done?
* System
* Unit

### CODE REVIEW

**Comment by texasaggie97** [comment]
test this please

**Comment by texasaggie97** [comment]
test this please

File: generated/nifake/nifake/session.py [hunk 1]
```diff
@@ -1295,7 +1297,7 @@ def import_attribute_configuration_buffer(self, configuration):
         '''
         vi_ctype = _visatype.ViSession(self._vi)  # case S110
         size_in_bytes_ctype = _visatype.ViInt32(0 if configuration is None else len(configuration))  # case S160
-        configuration_ctype = get_ctypes_pointer_for_buffer(value=_converters.convert_import_buffer_to_array(configuration), library_type=_visatype.ViInt8)  # case B520
+        configuration_ctype = get_ctypes_pointer_for_buffer(value=_converters.convert_to_bytes(configuration), library_type=_visatype.ViInt8)  # case B520
```
Comment (marcoskirsch) [inline generated/nifake/nifake/session.py]: In all the other drivers, export_attribute_configuration changes but not in nifake. Are we missing something?

File: generated/nifake/nifake/unit_tests/test_session.py [hunk 2]
```diff
@@ -1308,12 +1319,12 @@ def test_import_attribute_configuration_buffer_list_i8_big(self):
                 try:
                     session.import_attribute_configuration_buffer(configuration)
                     assert False
-                except OverflowError:
+                except ValueError:
                     pass
 
-    def test_import_attribute_configuration_buffer_list_i8_big_float(self):
+    def test_import_attribute_configuration_buffer_list_i8_float(self):
```
Comment (marcoskirsch) [inline generated/nifake/nifake/unit_tests/test_session.py]: I question whether we should have explicit tests for calling `import ` with a list of floats. It seems like a very invalid use case even if it happens to work due to duck typing.

File: generated/nifgen/nifgen/_library_singleton.py [hunk 3]
```diff
@@ -13,8 +13,8 @@
 _instance = None
 _instance_lock = threading.Lock()
 _library_info = {'Linux': {'64bit': {'name': 'nifgen', 'type': 'cdll'}},
-                 'Windows': {'32bit': {'name': 'nifgen_32.dll', 'type': 'windll'},
-                             '64bit': {'name': 'nifgen_64.dll', 'type': 'cdll'}}}
+                 'Windows': {'32bit': {'name': 'niFgen_32.dll', 'type': 'windll'},
```
Comment (marcoskirsch) [inline generated/nifgen/nifgen/_library_singleton.py]: what's with all this diff noise?

Comment (texasaggie97) [reply generated/nifgen/nifgen/_library_singleton.py]: Because I updated the nifgen metadata from our internal source and this is what it has.

Comment (marcoskirsch) [reply generated/nifgen/nifgen/_library_singleton.py]: I don't see metadata among the files being changed in this PR.

Comment (marcoskirsch) [reply generated/nifgen/nifgen/_library_singleton.py]: Those metadata updates should have happened in their own separate PRs, no?

Comment (texasaggie97) [reply generated/nifgen/nifgen/_library_singleton.py]: I'm not sure why. There were multiple changes in the metadata files, and I see the differences in the PR

Comment (texasaggie97) [reply generated/nifgen/nifgen/_library_singleton.py]: I thought you made them when you updated nifgen.

**Pull request merged**